File: Oefenopgaven_en_Huiswerk/progs/Week_3_Les_2_Sheet_58.py
import cmath
import numpy as np
import matplotlib.pyplot as plt

def H(Omega):
    return (2 - 1/2*cmath.e**(-1j*Omega)) / (1 + 3/4*cmath.e**(-1j*Omega))

print('  Ω        H(Ω)      |H(Ω)| arg(H(Ω)')
for Omega in np.arange(0, cmath.pi+0.001, cmath.pi/4):
    print('{:.3f}'.format(Omega),'{:13.3f}'.format(H(Omega)), '{:6.3f}'.format(abs(H(Omega))), '{:6.3f}'.format(cmath.phase(H(Omega))))

Omega = np.linspace(0, cmath.pi, 100)
plt.title('Versterking')
plt.figure(1)
plt.ylim(0, 10.5);
plt.plot(Omega, np.abs(H(Omega)))
plt.xlabel('Ω (radialen/sample)')
plt.xticks([0, 0.25*cmath.pi, 0.5*cmath.pi, 0.75*cmath.pi, cmath.pi], ['0', '1/4 π', '1/2 π', '3/4 π', 'π'])
plt.ylabel('|H(Ω)|')
plt.minorticks_on()
plt.grid(which='major', linestyle='-')
plt.grid(which='minor', linestyle=':')

plt.figure(2)
plt.plot(Omega, np.angle(H(Omega)))
plt.title('Fasedraaiing')
plt.xlabel('Ω (radialen/sample)')
plt.xticks([0, 0.25*cmath.pi, 0.5*cmath.pi, 0.75*cmath.pi, cmath.pi], ['0', '1/4 π', '1/2 π', '3/4 π', 'π'])
plt.ylabel('arg(H(Ω)) (rad)')
plt.minorticks_on()
plt.grid(which='major', linestyle='-')
plt.grid(which='minor', linestyle=':')
plt.show()

File: Oefenopgaven_en_Huiswerk/progs/Week_3_Les_2_Sheet_58.m
Omega = 0:pi/100:pi;
H = (2 - 1/2*exp(-1i*Omega)) ./ (1 + 3/4*exp(-1i*Omega));

figure(1);
plot(Omega, abs(H));
title('Versterking')
axis([0 pi 0 max(abs(H))*1.05]);
xlabel('\Omega (radialen/sample)');
set(gca, 'XTick', [0:pi/4:pi]);
set(gca, 'XTickLabel', {'0';'1/4\pi';'1/2\pi';'3/4\pi';'\pi'});
ylabel('|H(\Omega)|');
grid on;
grid minor;

figure(2);
plot(Omega, angle(H));
title('Fasedraaiing')
axis([0 pi 0 max(angle(H))*1.05]);
xlabel('\Omega (radialen/sample)');
ylabel('arg(H(\Omega)) (rad)');
set(gca, 'XTick', [0:pi/4:pi]);
set(gca, 'XTickLabel', {'0';'1/4\pi';'1/2\pi';'3/4\pi';'\pi'});
grid on;
grid minor;
